lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -144,15 +141,15 @@ export default class BitcoinProcessor {
       config.valueTimeLockUpdateEnabled,
       BitcoinClient.convertBtcToSatoshis(config.valueTimeLockAmountInBitcoins), // Desired lock amount in satoshis
       BitcoinClient.convertBtcToSatoshis(valueTimeLockTransactionFeesInBtc),    // Txn Fees amount in satoshis
-      ProtocolParameters.maximumValueTimeLockDurationInBlocks                   // Desired lock duration in blocks
+      this.versionManager
     );
   }
 
   /**
    * Initializes the Bitcoin processor
    */
   public async initialize () {
-    await this.versionManager.initialize();
+    await this.versionManager.initialize(this.blockMetadataStore);
```

Comment:
Style: inject all dependencies from constructor.

Reply:
I think it's better to do it in initialize because `VersionManager` doesn't need an instance of the store, it only takes it and use it to construct `Feecalculator`.

Reply:
ok, I'm fine with moving config to init